bokehjs/src/lib/models/renderers/glyph_renderer.ts

```diff
@@ -162,7 +172,21 @@ export class GlyphRendererView extends DataRendererView {
     this.connect(this.model.data_source.selected.change, render)
     this.connect(this.model.data_source._select, render)
     if (this.hover_glyph != null)
-      this.connect(this.model.data_source.inspect, render)
+      this.connect(this.model.data_source.inspect, () => {
+        // XXX: hoping for the best, assuming no in-place mutation
+        const {inspected} = this.model.data_source
+        const current_inspected = {
+          indices: inspected.indices,
+          line_indices: inspected.line_indices,
+          multiline_indices: inspected.multiline_indices,
+          image_indices: inspected.image_indices,
+          selected_glyphs: inspected.selected_glyphs,
+        }
+        if (!is_equal(this._previous_inspected, current_inspected)) {
```

Comment:
Could this caching and comparison be moved to the `SelectionManager`, which would simply not emit an `inspect` event at all if there is no change? Just wondering if it's possible to avoid adding more code to `glyph_rednerer.ts`

Reply:
I considered that, but `inspect` signal carries the selection geometry, which I suppose may change even if indices don't. Additionally, there are multiple points of emit of this signal (graph renderer), which complicates matters. Thus I intentionally separated emitting of this signal, from detection whether rendering is actually necessary. In fact, it would be best not to rely on this signal for rendering, but on `change` signal on `selected`, but that's currently a silent assignment, and it wouldn't work anyway, due to how equality works in `setv()`.

Reply:
> carries the selection geometry, which I suppose may change even if indices don't. 

Ah right, agree further improvements will have to wait.